### PR TITLE
TST: DataFrame initialization with an empty DataFrame with dtype

### LINKED ISSUE #20654: DataFrame initialization with an empty DataFrame with dtype
#### Code
```python
>>> pd.DataFrame(pd.DataFrame())
Empty DataFrame
Columns: []
Index: []

>>> pd.DataFrame([], dtype='object')
Empty DataFrame
Columns: []
Index: []

>>> pd.DataFrame(pd.DataFrame(), dtype='object')
Traceback (most recent call last):

  File "<ipython-input-62-36914d1cfa60>", line 1, in <module>
    pd.DataFrame(pd.DataFrame(), dtype='object')

  File "c:\users\hans\appdata\local\programs\python\python36\lib\site-packages\pandas\core\frame.py", line 328, in __init__
    dtype=dtype, copy=copy)

  File "c:\users\hans\appdata\local\programs\python\python36\lib\site-packages\pandas\core\generic.py", line 175, in _init_mgr
    if len(mgr.blocks) > 1 or mgr.blocks[0].values.dtype != dtype:

IndexError: tuple index out of range
```
#### Problem description
Initializing a DataFrame from an empty DataFrame with dtype='object' throws an unexpected exception.

#### Expected Output
Would expect an empty DataFrame to be created.

<details>
#### Output of ``pd.show_versions()``
commit: None
python: 3.6.3.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en
LOCALE: None.None

pandas: 0.21.1
pytest: 3.5.0
pip: 9.0.3
setuptools: 38.5.2
Cython: None
numpy: 1.13.3
scipy: 1.0.0
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: 1.7.1
patsy: None
dateutil: 2.7.0
pytz: 2018.3
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.1.1
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 1.0.1
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by gcaria**
- [x] closes #20654
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

File: pandas/tests/frame/test_constructors.py [hunk 1]
```diff
@@ -1203,6 +1203,12 @@ def test_constructor_DataFrame(self, float_frame):
         df_casted = DataFrame(float_frame, dtype=np.int64)
         assert df_casted.values.dtype == np.int64
 
+    def test_constructor_empty_DataFrame(self):
+
```
Comment (jreback) [inline pandas/tests/frame/test_constructors.py:1207]:
```suggestion
# GH 20624
```

File: pandas/tests/frame/test_constructors.py [hunk 2]
```diff
@@ -1203,6 +1203,12 @@ def test_constructor_DataFrame(self, float_frame):
         df_casted = DataFrame(float_frame, dtype=np.int64)
         assert df_casted.values.dtype == np.int64
 
+    def test_constructor_empty_DataFrame(self):
+# GH 20624
```
Comment (jreback) [inline pandas/tests/frame/test_constructors.py:1207]:
```suggestion
        # GH 20624
```

File: pandas/tests/frame/test_constructors.py [hunk 3]
```diff
@@ -1203,6 +1203,12 @@ def test_constructor_DataFrame(self, float_frame):
         df_casted = DataFrame(float_frame, dtype=np.int64)
         assert df_casted.values.dtype == np.int64
 
+    def test_constructor_empty_DataFrame(self):
```
Comment (jreback) [inline pandas/tests/frame/test_constructors.py:1206]:
```suggestion
    def test_constructor_empty_dataframe(self):
```

**Comment by gcaria** [comment]
Thanks for the help @jreback. 

However, it seems that pandas classes get capitalized in the test names (at least in this file), see the test just above (`test_constructor_DataFrame`) and the various `test_constructor_Series_...`.

**Comment by alimcmaster1** [comment]
> Thanks for the help @jreback.
> 
> However, it seems that pandas classes get capitalized in the test names (at least in this file), see the test just above (`test_constructor_DataFrame`) and the various `test_constructor_Series_...`.

Thanks for the PR - should be lower case as Jeff suggest and per PEP8. Can you double check why CI is failing/merge master? This looks good to me otherwise :)

**Comment by gcaria** [comment]
Thanks @alimcmaster1.Rebased to master and now all tests have passed

**Comment by jreback** [comment]
thanks @gcaria

**Pull request merged**